packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -296,7 +340,13 @@ export default function __experimentalUseColors(
 			InspectorControlsColorPanel: (
 				<InspectorControlsColorPanel { ...wrappedColorPanelProps } />
 			),
-			BackgroundColorDetector,
+			ColorDetector,
 		};
-	}, [ attributes, setAttributes, detectedBackgroundColorRef.current, ...deps ] );
+	}, [
+		attributes,
+		setAttributes,
+		detectedBackgroundColorRef.current,
+		detectedColorRef.current,
+		...deps,
+	] );
 }
```

Comment:
The code of this hook is becoming complex, we should test it properly.

Reply:
#18672